File: golang-backend/internal/models/billing.go
package models

import "time"

// บิล/ใบเสร็จรับเงิน (db)
type Billing struct {
	ID                      uint      `gorm:"primaryKey" json:"id"`                       //อันนี้เป็น primary key ของตาราง Billing
	VisitID                 uint      `gorm:"not null" json:"visit_id"`                   //อันนี้เป็น foreign key ที่เชื่อมกับตาราง VisitRecord
	TotalAmount             float64   `gorm:"not null;default:0" json:"total_amount"`     //อันนี้เป็นจำนวนเงินรวมทั้งหมดของบิล
	DiscountFromEligibility float64   `gorm:"default:0" json:"discount_from_eligibility"` //คือส่วนลดที่ได้จากสิทธิ์การรักษา
	NetAmount               float64   `gorm:"not null;default:0" json:"net_amount"`       //คือจำนวนเงินสุทธิหลังจากหักส่วนลดแล้ว
	PaymentMethod           string    `json:"payment_method"`                             //คือวิธีการชำระเงิน เช่น เงินสด บัตรเครดิต หรือโอนเงิน
	PaymentStatus           string    `gorm:"default:'pending'" json:"payment_status"`    //คือสถานะการชำระเงิน เช่น pending, paid, หรือ failed
	ReceiptNumber           string    `gorm:"uniqueIndex" json:"receipt_number"`          //คือหมายเลขใบเสร็จรับเงินที่ไม่ซ้ำกัน
	CreatedAt               time.Time `json:"created_at"`                                 //คือเวลาที่บิลถูกสร้างขึ้น
	UpdatedAt               time.Time `json:"updated_at"`                                 //คือเวลาที่บิลถูกแก้ไขล่าสุด

	VisitRecord VisitRecord `gorm:"foreignKey:VisitID" json:"visit_record"` //คือข้อมูลการเยี่ยมชมที่เกี่ยวข้องกับบิลนี้
}

File: react-frontend/src/types/billing.ts
// บิล/ใบเสร็จรับเงิน (Billing Table DB Model - B6741990)
export interface BillingDB {
  id: number;
  visit_id: number;
  total_amount: number;
  discount_from_eligibility: number;
  net_amount: number;
  payment_method: string;
  payment_status: 'pending' | 'paid' | 'cancelled';
  receipt_number?: string;
  created_at?: string;
  updated_at?: string;
}

// QR Code สำหรับชำระเงิน (QRPayment Table DB Model - B6741990)
export interface QRPaymentDB {
  id: number;
  billing_id: number;
  qr_code_data: string;
  promptpay_id: string;
  amount: number;
  status: 'pending' | 'completed' | 'expired';
  expired_at: string;
  created_at?: string;
  updated_at?: string;
}
